umap/static/umap/js/umap.layer.js

```diff
@@ -594,6 +596,31 @@ U.DataLayer = L.Evented.extend({
     if (this.autoLoaded()) this.map.on('zoomend', this.onZoomEnd, this)
   },
 
+  render: function (fields, builder) {
```

Comment:
Maybe we could call the method `onDataChanged` to make it more explicit that this is not the default redering but a sort of "reflow" ?

Reply:
I would prefer to keep the `on*` methods for events. (`onDataChanged` would imply there is an `data-changed` event that occurred). 

I like the idea of using `render` somewhere, because it's really the intent here. Maybe `rerenderFields`? (it also  makes it more explicit that's it's expecting fields to be passed).

Reply:
ok, let's keep `render`!